cmc-rnaseq-de/MSSM-Penn-Pitt_ACC_DLPFC_FEATURECOUNTS_GRCh38_CQN_Sparse.Rmd

```diff
@@ -841,20 +843,49 @@ gc()
+  glue::glue(" + (1|{random_effect})")
+
+L1 = variancePartition::getContrast(exprObj = VOOM.GENE_EXPRESSION, 
+                                   formula = formula, 
+                                   data = COVARIATES, 
+                                   coefficient = c("Dx.TissueOther_ACC", "Dx.TissueControl_ACC"))
+L2 = variancePartition::getContrast(exprObj = VOOM.GENE_EXPRESSION, 
+                                   formula = formula, 
+                                   data = COVARIATES, 
+                                   coefficient = c("Dx.TissueSCZ_ACC", "Dx.TissueControl_ACC"))
+L3 = variancePartition::getContrast(exprObj = VOOM.GENE_EXPRESSION, 
+                                   formula = formula, 
+                                   data = COVARIATES, 
+                                   coefficient = c("Dx.TissueSCZ_ACC", "Dx.TissueOther_ACC"))
+L4 = variancePartition::getContrast(exprObj = VOOM.GENE_EXPRESSION, 
+                                   formula = formula, 
+                                   data = COVARIATES, 
+                                   coefficient = c("Dx.TissueOther_DLPFC", "Dx.TissueControl_DLPFC"))
+L5 = variancePartition::getContrast(exprObj = VOOM.GENE_EXPRESSION, 
+                                   formula = formula, 
+                                   data = COVARIATES, 
+                                   coefficient = c("Dx.TissueSCZ_DLPFC", "Dx.TissueControl_DLPFC"))
+
+L6 = variancePartition::getContrast(exprObj = VOOM.GENE_EXPRESSION, 
+                                   formula = formula, 
+                                   data = COVARIATES, 
+                                   coefficient = c("Dx.TissueSCZ_DLPFC", "Dx.TissueOther_DLPFC"))
+L = cbind(L1, L2, L3, L4, L5, L6)
+
+# Visualize contrast matrix
+plotContrasts(L) 
+
 # Fit contrast
-contrast = makeContrasts(contrasts=c("Dx.TissueOther_ACC-Dx.TissueControl_ACC",
-                                     "Dx.TissueSCZ_ACC-Dx.TissueControl_ACC",
-                                     "Dx.TissueSCZ_ACC-Dx.TissueOther_ACC",
-                                     "Dx.TissueOther_DLPFC-Dx.TissueControl_DLPFC",
-                                     "Dx.TissueSCZ_DLPFC-Dx.TissueControl_DLPFC",
-                                     "Dx.TissueSCZ_DLPFC-Dx.TissueOther_DLPFC"),
-                         levels = colnames(ADJUSTED.FIT$coefficients))
-FIT.CONTR = contrasts.fit(ADJUSTED.FIT, contrasts=contrast)
-FIT.CONTR = eBayes(FIT.CONTR)
-
-# Get differnetial expression
-DE = lapply(1:6, function(i, FIT){
-  topTable(FIT, coef=i, number = 50000, confint = TRUE) %>%
+FIT.CONTR = variancePartition::dream(exprObj = VOOM.GENE_EXPRESSION,
+                                     formula = formula,
+                                     data = COVARIATES,
+                                     L = L)
+
+# Get differential expression
+DE = lapply(glue::glue("L{1:6}"), function(i, FIT){
+  topTable(FIT, coef=i, number = 50000) %>%
```

Comment:
I set this to number=Inf, because 50k is arbitrary.  This is fine for genes, by there are a larger number of transcripts, so might as well drop the cutoff.

Reply:
Great - will do.